# 01 - Data Gathering

### Notebook Overview

Load the car price dataset and perform an initial inspection to understand what we're working with before any cleaning or transformation.

**Dataset:** [Car Price Assignment](https://www.kaggle.com/datasets/imgowthamg/car-price) (Kaggle)  
**Task:** Regression (predict `price`)  
**Rows:** 205 | **Columns:** 26

### 1 - Imports

In [1]:
# Core Libraries
import pandas as pd

### 2 - Load Data

In [2]:
df = pd.read_csv("../data/car-price-assignment.csv")
print(f"Shape: {df.shape[0]} rows, {df.shape[1]} columns")
df.head()

Shape: 205 rows, 26 columns


,car_ID,symboling,CarName,fueltype,aspiration,doornumber,carbody,drivewheel,enginelocation,wheelbase,...,enginesize,fuelsystem,boreratio,stroke,compressionratio,horsepower,peakrpm,citympg,highwaympg,price
0,1,3,alfa-romero giulia,gas,std,two,convertible,rwd,front,88.6,...,130,mpfi,3.47,2.68,9.0,111,5000,21,27,13495.0
1,2,3,alfa-romero stelvio,gas,std,two,convertible,rwd,front,88.6,...,130,mpfi,3.47,2.68,9.0,111,5000,21,27,16500.0
2,3,1,alfa-romero Quadrifoglio,gas,std,two,hatchback,rwd,front,94.5,...,152,mpfi,2.68,3.47,9.0,154,5000,19,26,16500.0
3,4,2,audi 100 ls,gas,std,four,sedan,fwd,front,99.8,...,109,mpfi,3.19,3.40,10.0,102,5500,24,30,13950.0
4,5,2,audi 100ls,gas,std,four,sedan,4wd,front,99.4,...,136,mpfi,3.19,3.40,8.0,115,5500,18,22,17450.0


### 3 - Data Types and Structure

In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 205 entries, 0 to 204
Data columns (total 26 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   car_ID            205 non-null    int64  
 1   symboling         205 non-null    int64  
 2   CarName           205 non-null    str    
 3   fueltype          205 non-null    str    
 4   aspiration        205 non-null    str    
 5   doornumber        205 non-null    str    
 6   carbody           205 non-null    str    
 7   drivewheel        205 non-null    str    
 8   enginelocation    205 non-null    str    
 9   wheelbase         205 non-null    float64
 10  carlength         205 non-null    float64
 11  carwidth          205 non-null    float64
 12  carheight         205 non-null    float64
 13  curbweight        205 non-null    int64  
 14  enginetype        205 non-null    str    
 15  cylindernumber    205 non-null    str    
 16  enginesize        205 non-null    int64  
 17  fuelsyst

In [4]:
df.describe()

,car_ID,symboling,wheelbase,carlength,carwidth,carheight,curbweight,enginesize,boreratio,stroke,compressionratio,horsepower,peakrpm,citympg,highwaympg,price
count,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000,205.000000
mean,103.000000,0.834146,98.756585,174.049268,65.907805,53.724878,2555.565854,126.907317,3.329756,3.255415,10.142537,104.117073,5125.121951,25.219512,30.751220,13276.710571
std,59.322565,1.245307,6.021776,12.337289,2.145204,2.443522,520.680204,41.642693,0.270844,0.313597,3.972040,39.544167,476.985643,6.542142,6.886443,7988.852332
min,1.000000,-2.000000,86.600000,141.100000,60.300000,47.800000,1488.000000,61.000000,2.540000,2.070000,7.000000,48.000000,4150.000000,13.000000,16.000000,5118.000000
25%,52.000000,0.000000,94.500000,166.300000,64.100000,52.000000,2145.000000,97.000000,3.150000,3.110000,8.600000,70.000000,4800.000000,19.000000,25.000000,7788.000000
50%,103.000000,1.000000,97.000000,173.200000,65.500000,54.100000,2414.000000,120.000000,3.310000,3.290000,9.000000,95.000000,5200.000000,24.000000,30.000000,10295.000000
75%,154.000000,2.000000,102.400000,183.100000,66.900000,55.500000,2935.000000,141.000000,3.580000,3.410000,9.400000,116.000000,5500.000000,30.000000,34.000000,16503.000000
max,205.000000,3.000000,120.900000,208.100000,72.300000,59.800000,4066.000000,326.000000,3.940000,4.170000,23.000000,288.000000,6600.000000,49.000000,54.000000,45400.000000


In [5]:
df.describe(include="str")

,CarName,fueltype,aspiration,doornumber,carbody,drivewheel,enginelocation,enginetype,cylindernumber,fuelsystem
count,205,205,205,205,205,205,205,205,205,205
unique,147,2,2,2,5,3,2,7,7,8
top,peugeot 504,gas,std,four,sedan,fwd,front,ohc,four,mpfi
freq,6,185,168,115,96,120,202,148,159,94


### 4 - Missing Values

In [6]:
missing = df.isnull().sum()
missing[missing > 0] if missing.any() else print("No missing values")

No missing values


### 5 - Target Variable

In [7]:
print(f"Target: price")
print(f"  Min:    {df['price'].min():,.0f}")
print(f"  Max:    {df['price'].max():,.0f}")
print(f"  Mean:   {df['price'].mean():,.0f}")
print(f"  Median: {df['price'].median():,.0f}")
print(f"  Std:    {df['price'].std():,.0f}")

Target: price
  Min:    5,118
  Max:    45,400
  Mean:   13,277
  Median: 10,295
  Std:    7,989


### 6 - Column Overview

Quick reference for what each column represents.

In [8]:
col_summary = pd.DataFrame({
    "dtype": df.dtypes,
    "nunique": df.nunique(),
    "sample": df.iloc[0]
})
col_summary

,dtype,nunique,sample
car_ID,int64,205,1
symboling,int64,6,3
CarName,str,147,alfa-romero giulia
fueltype,str,2,gas
aspiration,str,2,std
doornumber,str,2,two
carbody,str,5,convertible
drivewheel,str,3,rwd
enginelocation,str,2,front
wheelbase,float64,53,88.6


### 7 - Observations

- **No missing values** across all 205 rows.
- **Target (price):** right-skewed. Mean ($13,277) is well above median ($10,295), range $5,118 to $45,400. Consider log-transforming the target during preprocessing.
- **`car_ID`:** row index only (1-205). Drop it.
- **`CarName`:** 147 unique values, high cardinality. Contains typos (e.g. "alfa-romero" vs "alfa romeo") and inconsistent casing. Extract brand/make as a new feature.
- **`symboling`:** insurance risk rating (-2 to 3). Integer but ordinal/categorical in nature.
- **`doornumber` and `cylindernumber`:** stored as text ("two", "four") instead of numeric. Convert during cleaning.
- **`enginelocation`:** 202 of 205 values are "front", near-zero variance. Candidate for removal.
- **Numeric columns:** no obvious garbage values from `describe()`; ranges look plausible for car specifications.